File: client/src/lib/summarizer.js
// lib/summarizer.js
function summarizeText(text) {
  return fetch('http://localhost:5000/summarizer/summarize', {
    method: 'POST',
    headers: {
      'Content-Type': 'application/json',
    },
    body: JSON.stringify({ text: text }),
  })
  .then(response => response.json())
  .then(data => data.summary);
}

File: server/app/api/summarizer.py
from flask import Blueprint, request, jsonify
from app.models.summarization_model import summarize_text

summarizer_blueprint = Blueprint('summarizer', __name__, url_prefix='/summarizer')

@summarizer_blueprint.route('/summarize', methods=['POST'])
def summarize():
    content = request.json
    text = content.get('text')
    summary = summarize_text(text)
    return jsonify(summary=summary)
